processor/src/test/java/org/mapstruct/ap/test/bugs/_2839/CarMapper.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2839;
+
+import java.util.UUID;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Hakan Özkan
+ */
+@Mapper
+public abstract class CarMapper {
+
+    public static final CarMapper MAPPER = Mappers.getMapper( CarMapper.class );
+
+    public abstract Car toEntity(CarDto dto);
+
+    public abstract CarDto toDto(Car entity);
+
+    protected Id mapId(UUID id) throws InstantiationException {
+        return id != null ? new Id( id ) : null;
+    }
+
+    protected UUID mapId(Id id) throws NoSuchMethodException {
+        return id != null ? id.getId() : null;
```

Comment:
It would be be nice to trigger the exception based on the data and verify that.

Reply:
I've removed `throws` from this method as it's not needed here.

Reply:
It is not only for removing the `throws`. I think what @sjaakd wanted to see was to throw an exception in one of these methods and verify that in the tests.

Reply:
Oh, I see. Done, the exceptions are thrown now. One additional thing that I noticed is that checked exceptions are failing however unchecked exceptions are just fine.